ENH: Add no cross terms option to White's test for heteroscedasticity

IntegralIndefinida opened the pull request:
This PR adds a `interaction_terms` parameter to the `het_white` function to allow users to choose whether to include interaction terms in White's heteroskedasticity test. This is useful since adding the cross terms consumes degrees of freedom. Additionally, White's heteroscedasticity test with cross terms can also be a specification test, and, according to Richard Harris (as cited in Gujarati's chapter on heteroscedasticity), if we remove the cross terms, it constitutes a pure heteroscedasticity test. 

**Changes**

- Added `interaction_terms` parameter (default `True`) to `statsmodels.stats.diagnostic.het_white`
- When `interaction_terms=False`, the test uses only squared terms (x1², x2², ...) without interaction terms (x1*x2, x1*x3, ...)

**Tests**

- Added `test_het_white_no_interaction_terms` to verify the `interaction_terms=False` option
- Reference values verified against EViews (I also verified the base variant results)

- [ x ] tests added / passed. 
- [ x ] code/documentation is well formatted.  
- [ x ] properly formatted commit message. See 
      [NumPy's guide](https://docs.scipy.org/doc/numpy-1.15.1/dev/gitwash/development_workflow.html#writing-the-commit-message). 

<details>


**Notes**:

* It is essential that you add a test when making code changes. Tests are not 
  needed for doc changes.
* When adding a new function, test values should usually be verified in another package (e.g., R/SAS/Stata).
* When fixing a bug, you must add a test that would produce the bug in main and
  then show that it is fixed with the new code.
* New code additions must be well formatted. Changes should pass flake8. If on Linux or OSX, you can
  verify you changes are well formatted by running 
  ```
  git diff upstream/main -u -- "*.py" | flake8 --diff --isolated
  ```
  assuming `flake8` is installed. This command is also available on Windows 
  using the Windows System for Linux once `flake8` is installed in the 
  local Linux environment. While passing this test is not required, it is good practice and it help 
  improve code quality in `statsmodels`.
* Docstring additions must render correctly, including escapes and LaTeX.

</details>

josef-pkt commented:
Looks good overall

I guess this will then be equivalent to 
`het_breuschpagan(resid, exog**2)`

So, not really needed but addition is fine with me.

josef-pkt commented:
Thanks,
PR looks good.
Waiting for the CI to finish, but it looks to me it's ready for merging.

josef-pkt commented:
ci fails
```
>       hw = smdia.het_white(res.resid, res.model.exog, interaction_terms=False)
             ^^^^^
E       NameError: name 'smdia' is not defined
```

and one style failure with white space

bashtage commented on statsmodels/stats/diagnostic.py:
```diff
@@ -835,8 +835,10 @@ def het_white(resid, exog):
-        terms are automatically included in the auxiliary regression.
+        The explanatory variables for the variance. Squares and, by default,
+        interaction terms are automatically included in the auxiliary regression.
+    interaction_terms : bool, default True
```
Should probably be False for now since this would change the output of tests without a deprecation.

IntegralIndefinida commented:
I fixed the missing line, but the test still fails with the following linting errors:
~~~ 
Running flake8 linting
Linting all files with limited rules
statsmodels/discrete/discrete_model.py:522:13: B043 Do not call delattr with a constant attribute value, it is not any safer than normal property access.
statsmodels/discrete/discrete_model.py:1054:13: B043 Do not call delattr with a constant attribute value, it is not any safer than normal property access.
statsmodels/discrete/discrete_model.py:1056:13: B043 Do not call delattr with a constant attribute value, it is not any safer than normal property access.
statsmodels/genmod/generalized_linear_model.py:375:13: B043 Do not call delattr with a constant attribute value, it is not any safer than normal property access.
statsmodels/genmod/generalized_linear_model.py:377:13: B043 Do not call delattr with a constant attribute value, it is not any safer than normal property access.
Changed files failed linting using the required set of rules.
Additions and changes must conform to Python code style rules.
No new files to lint
Running isort
Skipped 1 files

##[error]Bash exited with code '1'.
Finishing: Check style

~~~
Those are unrelated to my commits

bashtage commented:
Thanks.
